src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -54,6 +54,50 @@ private void processMatchingMilestonesByQueryWord(List<PickerMilestone> mileston
     }
 
     /**
+     * Selects the next milestone in the list of best matching milestones
+     */
+    public void selectNextBestMatchingMilestone() {
+        int curSelectedIndex = getSelectedIndex(bestMatchingMilestones);
+        if (curSelectedIndex >= bestMatchingMilestones.size() - 1) return;
+        selectMilestoneAtIndex(bestMatchingMilestones, curSelectedIndex + 1);
+    }
+
+    /**
+     * Selects the previous milestone in the list of best matching milestones
+     */
+    public void selectPreviousBestMatchingMilestone() {
+        int curSelectedIndex = getSelectedIndex(bestMatchingMilestones);
+        if (curSelectedIndex <= 0) return;
+        selectMilestoneAtIndex(bestMatchingMilestones, curSelectedIndex - 1);
+    }
+
+    /**
+     * Returns the first index at which a milestone is selected
+     *
+     * @param milestones
+     * @return -1 if there are no selected milestones
+     */
+    private int getSelectedIndex(List<PickerMilestone> milestones) {
+        for (int i = 0; i < milestones.size(); i++) {
+            if (milestones.get(i).isSelected()) {
+                return i;
+            }
+        }
+        return -1;
```

Comment:
Is it possible that nothing is selected? If so, the return type should be optional (-1 is kind of ad hoc). If it's not possible, there should be an assertion failure.